modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/function/MBFunction_toColor.java

```diff
@@ -0,0 +1,140 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2018, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.mbstyle.function;
+
+import org.geotools.filter.FunctionExpressionImpl;
+import org.geotools.filter.capability.FunctionNameImpl;
+import org.json.simple.JSONArray;
+import org.opengis.filter.capability.FunctionName;
+import org.opengis.filter.expression.Expression;
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Takes an object as an argument and returns the color value if possible.
+ * Evaluates string in the formats of:
+ * "rgb(int, int, int)"
+ * "rgba(int, int, int, double)"
+ *
+ * And in arrays of 3 and 4 numbers for rgb and rgba colors:
+ * [int, int, int]
+ * [int, int, int, double]
+ *
+ * The integer values should be between 0-255, for red, green, and blue color values.
+ * The double value should be between 0-1, and is converted to a 0-255 alpha value.
+ */
+class MBFunction_toColor extends FunctionExpressionImpl {
+
+    public static FunctionName NAME = new FunctionNameImpl("mbToColor");
+
+    MBFunction_toColor() {
+        super(NAME);
+    }
+
+    /**
+     * @see org.geotools.filter.FunctionExpressionImpl#setParameters(java.util.List)
+     */
+    @Override
+    public void setParameters(List<Expression> params) {
+        // set the parameters
+        this.params = new ArrayList<>(params);
+    }
+
+    @Override
+    public Object evaluate(Object feature) {
+        Color c;
+        for (Integer i = 1; i <= this.params.size() - 1; i++) {
+            Object evaluation = this.params.get(i).evaluate(feature);
+            if (evaluation instanceof Color) {
+                return evaluation;
+            }
+            if (evaluation instanceof JSONArray) {
+                JSONArray je = (JSONArray) evaluation;
+                if (je.size() == 3 || je.size() == 4) {
```

Comment:
Think we made similar logic for processing of color constants, can some thought be given to avoiding duplication of logic.

Reply:
I believe some of that logic was in the YSLD extension (specifically `#RGB` colors), so duplicating that can't really be avoided (You could pull it out to a util class, but both work slightly differently I think). Any duplication that can be avoided would be good though.